sdk/src/consumer_groups/delete_consumer_group.rs

```diff
@@ -33,30 +34,30 @@ impl Validatable<IggyError> for DeleteConsumerGroup {
 }
 
 impl BytesSerializable for DeleteConsumerGroup {
-    fn as_bytes(&self) -> Vec<u8> {
+    fn as_bytes(&self) -> Bytes {
         let stream_id_bytes = self.stream_id.as_bytes();
         let topic_id_bytes = self.topic_id.as_bytes();
         let consumer_group_id_bytes = self.consumer_group_id.as_bytes();
-        let mut bytes = Vec::with_capacity(
+        let mut bytes = BytesMut::with_capacity(
             stream_id_bytes.len() + topic_id_bytes.len() + consumer_group_id_bytes.len(),
         );
-        bytes.extend(stream_id_bytes);
-        bytes.extend(topic_id_bytes);
-        bytes.extend(consumer_group_id_bytes);
-        bytes
+        bytes.put_slice(&stream_id_bytes);
+        bytes.put_slice(&topic_id_bytes);
+        bytes.put_slice(&consumer_group_id_bytes);
+        bytes.freeze()
     }
 
-    fn from_bytes(bytes: &[u8]) -> Result<DeleteConsumerGroup, IggyError> {
+    fn from_bytes(bytes: Bytes) -> Result<DeleteConsumerGroup, IggyError> {
         if bytes.len() < 9 {
             return Err(IggyError::InvalidCommand);
         }
 
         let mut position = 0;
-        let stream_id = Identifier::from_bytes(bytes)?;
+        let stream_id = Identifier::from_bytes(bytes.clone())?;
```

Comment:
like above, consider changing signature of from_bytes to accept ref and perhaps try to pass a reference

Reply:
https://docs.rs/bytes/1.5.0/bytes/struct.Bytes.html

ref：`They are cheaply cloneable and thereby shareable between an unlimited amount of components, for example by modifying a reference count.`

https://docs.rs/bytes/1.5.0/bytes/struct.Bytes.html#sharing

Bytes clone and passing &Bytes incur almost the same cost, it doesn't actually copy the underlying data, just increases the reference count.